server/routes/user.js

```diff
@@ -1,77 +1,34 @@
 const express = require('express');
 const router = express.Router();
+const { ApiService } = require('../services');
 const User = require('../database/models/user');
-const passport = require('../passport');
 
-router.post('/', (req, res) => {
-    console.log('user signup');
+router.get('/', async (req, res, next) => {
+  // validate auth token (return 401 if invalid)
+  const userInfoResponse = await ApiService.get(`${process.env.AWS_COGNITO_BASE_URL}/oauth2/userInfo`, {
+    headers: { Authorization: `${req.get('Authorization')}` },
+  }).catch((error) => {
+    res.sendStatus(401);
+    next(JSON.stringify(error.response.data));
+  });
 
-    const { username, password, name, email } = req.body;
-    // ADD VALIDATION
-    User.findOne({ username: username }, (err, user) => {
-        if (err) {
-            console.log('User.js post error: ', err);
-        } else if (user) {
-            res.json({
-                error: `Sorry, already a user with the username: ${username}`,
-            });
-        } else {
-            const newUser = new User({
-                username: username,
-                password: password,
-                name: name,
-                email: email,
-            });
-            newUser.save((err, savedUser) => {
-                if (err) return res.json(err);
-                res.json(savedUser);
-            });
-        }
-    });
-});
-
-router.get('/data/:username', (req, res, next) => {
-    // console.log("this is for user: " + req.params.username)
-    User.findOne({ username: req.params.username }).then(
-        (userData) => res.json(userData),
-        (error) => res.sendStatus(404)
-    );
-});
+  // if authorization fails, exit function
+  if (!userInfoResponse) return;
```

Comment:
@LGG233 Okay, so I learned that `res.sendStatus(401)` causes the API Response to fire off but it _does not_ cause the javascript route handler function to cease execution. So, I added a null check and function return in the case that authorization fails.

Reply:
Good catch 👍